File: examples/io_grpc/bazel_build_java/service/armeria/src/main/kotlin/com/geekinasuit/daggergrpc/iogrpc/example/armeria/services/WhateverServiceAdapter.java
package com.geekinasuit.daggergrpc.iogrpc.example.armeria.services;

import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.Whatever;
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.WhateverServiceGrpc;
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.WhateverServiceGrpc.AsyncService;
import io.grpc.BindableService;
import io.grpc.ServerServiceDefinition;
import io.grpc.stub.StreamObserver;
import java.util.concurrent.Callable;
import javax.annotation.processing.Generated;

@Generated("to be generated")
public class WhateverServiceAdapter implements BindableService, AsyncService {
  private final Callable<AsyncService> service;

  public WhateverServiceAdapter(Callable<AsyncService> service) {
    this.service = service;
  }

  @Override
  public void whatever(
      Whatever.WhateverRequest req, StreamObserver<Whatever.WhateverResponse> resp) {
    try {
      service.call().whatever(req, resp);
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }

  @Override
  public ServerServiceDefinition bindService() {
    return WhateverServiceGrpc.bindService(this);
  }
}

File: examples/io_grpc/bazel_build_kt/service/armeria/src/main/kotlin/com/geekinasuit/daggergrpc/iogrpc/example/armeria/services/WhateverServiceAdapter.kt
package com.geekinasuit.daggergrpc.iogrpc.example.armeria.services

import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.Whatever.WhateverRequest
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.Whatever.WhateverResponse
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.WhateverServiceGrpc
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.WhateverServiceGrpc.AsyncService
import io.grpc.BindableService
import io.grpc.ServerServiceDefinition
import io.grpc.stub.StreamObserver
import javax.annotation.Generated

@Generated("to be generated")
class WhateverServiceAdapter(val service: () -> AsyncService) : BindableService, AsyncService {
  override fun whatever(req: WhateverRequest, resp: StreamObserver<WhateverResponse>) =
    service().whatever(req, resp)

  override fun bindService(): ServerServiceDefinition = WhateverServiceGrpc.bindService(this)
}
